fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/Client.java

```diff
@@ -229,7 +232,7 @@ public static Client createNew(final AppUser currentUser, final Office clientOff
             final Integer legalForm, final JsonCommand command) {
 
         final String accountNo = command.stringValueOfParameterNamed(ClientApiConstants.accountNoParamName);
-        final String externalId = command.stringValueOfParameterNamed(ClientApiConstants.externalIdParamName);
+        final ExternalId externalId = ExternalId.empty();
```

Comment:
It cannot be empty, you should fetch it from the command and if the configuration is enabled, generate new one. It worth to extract this logic out from the entity. Use a factory / producer to get the data from the command and call the proper constructor.

Reply:
Done!